Inferences/ForwardDemodulation.cpp

```diff
@@ -194,10 +200,7 @@ bool ForwardDemodulationImpl<combinatorySupSupport>::perform(Clause* cl, Clause*
           }
         }
 
-        TermList rhsS = subs->applyToBoundResult(rhs);
-        if (lhs.isVar()) {
-          rhsS = eqSortSubs.apply(rhsS, 0);
-        }
+        TermList rhsS = rhsApplied.apply();
```

Comment:
Note: I changed this because we already have substitution-applying logic in the applicator classes above, but we can change it back if it shows unexpected slowdown. (I measured it only superficially so far.)

Reply:
I didn't find any change in performance just now...